sklearn/semi_supervised/label_propagation.py

```diff
@@ -272,8 +278,9 @@ class LabelPropagation(BaseLabelPropagation):
 
     Parameters
     ----------
-    kernel : {'knn', 'rbf'}
-        String identifier for kernel function to use.
+    kernel : {'knn', 'rbf', function}
+        String identifier for kernel function to use or the kernel function to calculate
+        the [n_samples, n_samples] sized weight matrix.
```

Comment:
Can you be more explicit about the expected input and output values of the user-provided function?